components/Form/Register.tsx

```diff
@@ -0,0 +1,128 @@
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import React, { useState } from "react";
+import { View, Text, TextInput, Pressable, Alert } from "react-native";
+
+interface FormProps {
+  onNavigate: () => void;
+}
+
+export default function Register({ onNavigate }: FormProps) {
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [email, setEmail] = useState("");
+
+  const checkExistingUser = async (userEmail: string) => {
+    try {
+      const existingUsers = await AsyncStorage.getItem("registeredUsers");
+      if (existingUsers) {
+        const users = JSON.parse(existingUsers);
+        return users.some(
+          (user: { email: string }) => user.email === userEmail
+        );
+      }
+      return false;
+    } catch (error) {
+      console.error("Failed to fetch the users", error);
+      return false;
+    }
+  };
+
+  const saveRegisterData = async () => {
+    try {
+      const userExists = await checkExistingUser(email);
+
+      if (userExists) {
+        Alert.alert("Error", "User with this email already exists");
+        return false;
+      }
+
+      const existingUsers = await AsyncStorage.getItem("registeredUsers");
+      const users = existingUsers ? JSON.parse(existingUsers) : [];
+      users.push({ name, email, password });
+
+      await AsyncStorage.setItem("registeredUsers", JSON.stringify(users));
+      return true;
+    } catch (error) {
+      console.error("Failed to fetch Data", error);
```

Comment:
Errors are logged but not presented to users when saveRegisterData fails. Add `Alert.alert` or appropriate feedback for error scenarios.